gulpfile.js

```diff
@@ -6,44 +6,42 @@ var istanbul = require('gulp-istanbul');
 var plumber = require('gulp-plumber');
 var babel = require('gulp-babel');
 
-// Initialize the babel transpiler so ES2015 files gets compiled
-// when they're loaded
-require('babel-core/register');
-
 gulp.task('static', function () {
-  return gulp.src('**/*.js')
+  return gulp
+    .src('**/*.js')
     .pipe(excludeGitignore())
     .pipe(eslint())
     .pipe(eslint.format())
     .pipe(eslint.failAfterError());
 });
 
 gulp.task('pre-test', function () {
-  return gulp.src('lib/**/*.js')
+  return gulp
+    .src('lib/**/*.js')
     .pipe(babel())
-    .pipe(istanbul({includeUntested: true}))
+    .pipe(istanbul({ includeUntested: true }))
     .pipe(istanbul.hookRequire());
 });
 
-gulp.task('test', ['pre-test'], function (cb) {
-  var mochaErr;
-
-  gulp.src('test/**/*.js')
-    .pipe(plumber())
-    .pipe(mocha({reporter: 'spec'}))
-    .on('error', function (err) {
-      mochaErr = err;
-    })
-    .on('end', function () {
-      cb(mochaErr);
-    });
-});
+gulp.task(
```

Comment:
https://medium.com/@defrian.yarfi/https-medium-com-defrian-yarfi-upgrade-gulp-3-to-gulp-4-the-gulpfile-js-workflow-45196e333b86